packages/docs/docs/docs/linking.md

```diff
@@ -1,127 +1,281 @@
+
+```console
+$ zos create
+? Pick a contract to instantiate: openzeppelin-eth/StandaloneERC20
+? Pick a network: development
+✓ Deploying openzeppelin-eth dependency to network
+? Do you want to call a function on the instance after creating it?: Yes
+? Select which function: * initialize(name: string, symbol: string, decimals: uint8, initialSupply: uint256, initialHolder: address, minters: address[], pausers: address[])
+? name (string): MyToken
+? symbol (string): MYT
+? decimals (uint8): 18
+? initialSupply (uint256): 100e18
+? initialHolder (address): 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
+? minters (address[]): 
+? pausers (address[]): 
+✓ Setting everything up to create contract instances
+✓ Instance created at 0x2612Af3A521c2df9EAF28422Ca335b04AdF3ac66
+```
+
+Let's break down what we did in the command above. We first chose to create an instance of the `StandaloneERC20` contract from the `openzeppelin-eth` package we had linked before, and to create it in the local `development` network. We are then instructing the CLI to _initialize_ it with the initial values needed to set up our token. This requires us to choose the appropriate `initialize` function, and input all the required arguments. ZeppelinOS will then atomically create and initialize the new instance in a single transaction.
+
+We now have a working ERC20 token contract in our development network. We can check that the initial supply was properly allocated by using the `balance` command. Make sure to use the address where your ERC20 token instance was created.
+
+<!-- CODE: We need a way to refer to our own contracts by name (and to the local accounts as well!) -->
+
+```console
+$ zos balance --erc20 0x2612Af3A521c2df9EAF28422Ca335b04AdF3ac66
+? Enter an address to query its balance: 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
+? Pick a network: development
+Balance: 100 MYT
+```
+
+Great! We can now write an exchange contract and connect it to this token when we deploy it.
+
+<!-- We can split the following into a separate tutorial if this one becomes too long -->
+
+## Writing the exchange contract
+
+Our exchange contract will need to store the token contract address and the exchange rate in its state. We will set these values during initialization, when we deploy our contract.
+
+In order to support contract upgrades, ZeppelinOS [does not allow the usage of Solidity's `constructor`s](pattern#the-constructor-caveat). Instead, we need to use _initializers_. An initializer is just a regular Solidity function, with an additional check to ensure that it can be called only once. To make coding initializers easy, ZeppelinOS provides a base `Initializable` contract, that includes an `initializer` modifier that takes care of this. You will first need to install the package that provides that contract:
+
+```console
+npm install zos-lib@2.4.0
+```
+
+Now, let's write our exchange contract using an _initializer_ to set its initial state:
```

Comment:
Should we add lines like 
"Create a TokenExchange.sol file using below code under contracts folder". Or we assume the reader will get it.